src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1039,6 +1039,45 @@ def create_or_replace_dynamic_table(
             source_plan,
         )
 
+    def _merge_file_format_options(
+        self, file_format_options: Dict[str, Any], options: Dict[str, str]
+    ) -> Dict[str, Any]:
+        """
+        Merges remotely defined file_format options with the local options set. This allows the client
+        to override any locally defined options that are incompatible with the file format.
+        """
+        if "FORMAT_NAME" not in options:
+            return file_format_options
+
+        def process_list(list_property):
+            split_list = list_property.lstrip("[").rstrip("]").split(", ")
+            if len(split_list) == 1:
+                return split_list[0]
+            return tuple(split_list)
+
+        type_map = {
+            "String": str,
+            "List": process_list,
+            "Integer": int,
+            "Boolean": bool,
+        }
+        new_options = {**file_format_options}
+        file_format = self.session.sql(
+            f"DESCRIBE FILE FORMAT {options['FORMAT_NAME']}"
```

Comment:
We may be careful about eagerly triggering such a query 1) it will be visible to users 2) maybe add a try/catch because it's still possible that it will fail.

Reply:
That's a good point. I think I can do this better in a way that maintains lazy evaluation. I'll refactor a bit.

Reply:
Can you create a jira and add a todo here for doing it lazily in the future?